xchange-deribit/src/main/java/org/knowm/xchange/deribit/v2/service/DeribitTradeService.java

```diff
@@ -1,20 +1,339 @@
+import org.knowm.xchange.deribit.v2.dto.Kind;
+import org.knowm.xchange.deribit.v2.dto.trade.*;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.Order.IOrderFlags;
+import org.knowm.xchange.dto.account.OpenPositions;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.MarketOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.StopOrder;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
+import org.knowm.xchange.instrument.Instrument;
 import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.*;
+import org.knowm.xchange.service.trade.params.orders.*;
 
 public class DeribitTradeService extends DeribitTradeServiceRaw implements TradeService {
 
   public DeribitTradeService(DeribitExchange exchange) {
     super(exchange);
   }
 
+  @Override
+  public OpenOrders getOpenOrders() throws IOException {
+    return DeribitAdapters.adaptOpenOrders(openOrders());
+  }
+
+  @Override
+  public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+    List<org.knowm.xchange.deribit.v2.dto.trade.Order> openOrders;
+
+    if (params instanceof OpenOrdersParamCurrencyPair) {
+      OpenOrdersParamCurrencyPair pairParams = (OpenOrdersParamCurrencyPair) params;
+      CurrencyPair pair = pairParams.getCurrencyPair();
+      openOrders = super.getOpenOrdersByCurrency(pair.base.getCurrencyCode(), null, null);
+    } else if (params instanceof OpenOrdersParamInstrument) {
+      OpenOrdersParamInstrument instrumentParams = (OpenOrdersParamInstrument) params;
+      Instrument instrument = instrumentParams.getInstrument();
+      openOrders =
+          super.getOpenOrdersByInstrument(DeribitAdapters.adaptInstrumentName(instrument), null);
+    } else {
+      openOrders = openOrders();
+    }
+
+    return DeribitAdapters.adaptOpenOrders(openOrders);
+  }
+
+  @Override
+  public OpenOrdersParams createOpenOrdersParams() {
+    return new DefaultOpenOrdersParamInstrument();
+  }
+
+  private List<org.knowm.xchange.deribit.v2.dto.trade.Order> openOrders() throws IOException {
+    try {
+      return ((DeribitAccountService) exchange.getAccountService())
+          .currencies().stream()
+              .map(Currency::getCurrencyCode)
+              .parallel()
+              .flatMap(
+                  c -> {
+                    try {
+                      return super.getOpenOrdersByCurrency(c, null, null).stream();
+                    } catch (IOException e) {
+                      throw new ExchangeException(e);
+                    }
+                  })
+              .collect(Collectors.toList());
+    } catch (ExchangeException e) {
+      if (e.getCause() instanceof IOException) {
+        throw (IOException) e.getCause();
+      }
```

Comment:
because of this the in this case the code would actually be more readable as a plain old loop

Reply:
Totally agree, now it's more readable